src/test/java/jakarta/validation/ValidationTest.java

```diff
@@ -165,6 +169,29 @@ private int countInMemoryProviders() {
 		}
 		return count;
 	}
+	
+	@Test
+	public void testWeakHashMapSynchronization() {
+		try {
+			Validation validation = new Validation();
+			GenericBootstrap f1 = Validation.byDefaultProvider();
+			System.out.println(f1.getClass().getDeclaredField("defaultResolver"));
+			Class<?>[] classes = validation.getClass().getDeclaredClasses();
+			for (int i = 0; i < classes.length; i++) {
+				if (classes[i].getSimpleName().equals("GetValidationProviderListAction")) {
+					Field field = classes[i].getDeclaredField("providersPerClassloader");
+					field.setAccessible(true);
+					Constructor<?> constructor = classes[i].getDeclaredConstructor();
+					constructor.setAccessible(true);
+					Object obj = constructor.newInstance();
+					assertTrue(field.get(obj).getClass().getName().equals("java.util.Collections$SynchronizedMap"));
+				}
+			}
+		} catch (Exception e) {
+			e.printStackTrace();
+			fail();
+		} 
+	}
```

Comment:
Thanks but you can drop the test, it really has no value. If we were doing that for everything, it would just be a big nightmare to maintain.

Reply:
okay i will remove the test